index.js

```diff
@@ -82,46 +93,47 @@ function serveStatic(root, options) {
       return
     }
 
-    var forwardError = !fallthrough
     var originalUrl = parseUrl.original(req)
-    var path = parseUrl(req).pathname
+    var originalPath = parseUrl(req).pathname
 
     // make sure redirect occurs at mount
-    if (path === '/' && originalUrl.pathname.substr(-1) !== '/') {
-      path = ''
+    if (originalPath === '/' && originalUrl.pathname.substr(-1) !== '/') {
+      originalPath = ''
     }
 
-    // create send stream
-    var stream = send(req, path, opts)
-
-    // add directory handler
-    stream.on('directory', onDirectory)
+    // options passed to stream
+    var streamOptions = {
+      opts: opts,
+      fallthrough: fallthrough,
+      setHeaders: setHeaders,
+      onDirectory: onDirectory,
+      path: originalPath
+    }
 
-    // add headers listener
-    if (setHeaders) {
-      stream.on('headers', setHeaders)
+    // static gzip serving disabled
+    var fallbackStream = streamFile.bind(this, req, res, next, streamOptions)
+    if (!serveGzip) {
+      return fallbackStream()
     }
 
-    // add file listener for fallthrough
-    if (fallthrough) {
-      stream.on('file', function onFile() {
-        // once file is determined, always forward error
-        forwardError = true
-      })
+    // gzip encoding not supported by client
+    var acceptEncoding = req.headers['accept-encoding'] || ''
+    if (acceptEncoding.indexOf('gzip') === -1) {
```

Comment:
This is not an acceptable way to do the content-negotiation, especially since Apache accepts a declaration `no-gzip` in this header, and this line will actually cause Gzip to occur in that case. Can we get this string fully parsed according to the HTTP specifications so we are not accidentally matching something incorrectly?